### DISCUSSION TITLE
Wrong locator passed into the server's list TCPv4:127.0.0.1:12345

### DISCUSSION THREAD

**Discussion by haavarpb** [eProsima/Fast-DDS#5358] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Hi, I am trying to set up ROS2 for a University class while keeping the IT dept. happy not using multicast (SIMPLE discovery).

I've tried to understand how to set up the discovery server locally on my machine first, before I test on the class robots. However, I am not able to understand how to do it. I simply want to run the pub/sub example using a local discovery server over TCP.

I've specified the `ROS_DISCOVERY_SERVER=TCPv4:127.0.0.1:12345`, launched `fastdds discovery -i 0` and then the pub/sub nodes.

I'm expecting that the server starts without faults, and listens on the localhost interface, on the specified port using TCP. I also expect that the pub-subs be able to communicate.

### Current behavior

fastdds discovery output:

```bash
2024-10-07 16:31:57.016 [SERVER_CLIENT_DISCOVERY Error] Wrong locator passed into the server's list TCPv4:127.0.0.1:12345 -> Function load_environment_server_info
### Server is running ###
  Participant Type:   SERVER
  Security:           NO
  Server ID:          0
  Server GUID prefix: 44.53.00.5f.45.50.52.4f.53.49.4d.41
  Server Addresses:   UDPv4:[0.0.0.0]:11811
```

Pub and sub terminals display the same error.

### Steps to reproduce

In first terminal:

```bash
export ROS_DISCOVERY_SERVER=TCPv4:127.0.0.1:12345
fastdds discovery -i 0
```

In second terminal
```bash
export ROS_DISCOVERY_SERVER=TCPv4:127.0.0.1:12345
ros2 run examples_rclpy_minimal_publisher publisher_old_school
```

In third terminal

```bash
export ROS_DISCOVERY_SERVER=TCPv4:127.0.0.1:12345
ros2 run examples_rclpy_minimal_subscriber subscriber_old_school
```

### Fast DDS version/commit

apt list | egrep "fastdds|*humble.*fastrtps"

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

fastdds-tools/jammy 2.5.0+ds-3 amd64
fastddsgen/jammy,jammy 2.1.0+dfsg-2 all
ros-humble-fastrtps-cmake-module/jammy,now 2.2.2-2jammy.20240728.202329 amd64 [installed,automatic]
ros-humble-fastrtps-dbgsym/jammy 2.6.8-1jammy.20240728.201342 amd64
ros-humble-fastrtps/jammy,now 2.6.8-1jammy.20240728.201342 amd64 [installed,automatic]
ros-humble-rmw-fastrtps-cpp-dbgsym/jammy 6.2.7-1jammy.20240728.212513 amd64
ros-humble-rmw-fastrtps-cpp/jammy,now 6.2.7-1jammy.20240728.212513 amd64 [installed]
ros-humble-rmw-fastrtps-dynamic-cpp-dbgsym/jammy 6.2.7-1jammy.20240728.212514 amd64
ros-humble-rmw-fastrtps-dynamic-cpp/jammy 6.2.7-1jammy.20240728.212514 amd64
ros-humble-rmw-fastrtps-shared-cpp-dbgsym/jammy 6.2.7-1jammy.20240728.211440 amd64
ros-humble-rmw-fastrtps-shared-cpp/jammy,now 6.2.7-1jammy.20240728.211440 amd64 [installed,automatic]
ros-humble-rosidl-typesupport-fastrtps-c-dbgsym/jammy 2.2.2-2jammy.20240728.210059 amd64
ros-humble-rosidl-typesupport-fastrtps-c/jammy,now 2.2.2-2jammy.20240728.210059 amd64 [installed,automatic]
ros-humble-rosidl-typesupport-fastrtps-cpp-dbgsym/jammy 2.2.2-2jammy.20240728.205803 amd64
ros-humble-rosidl-typesupport-fastrtps-cpp/jammy,now 2.2.2-2jammy.20240728.205803 amd64 [installed,automatic]

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

TCPv4

### Additional context

ROS2 Humble, Ubuntu 22.04

**Comment by JesusPoderoso** [comment]
Hi @haavarpb, thanks for your report.
There are a couple of errors in the commands you've shared.

The wrong locator error is caused by the lack of brackets (`[`, `]`) around the IP address. This would be the correct export:
```bash
export ROS_DISCOVERY_SERVER=TCPv4:[127.0.0.1]:12345
```
Please, refer to the [`ROS_DISCOVERY_SERVER`](https://fast-dds.docs.eprosima.com/en/2.6.x/fastdds/env_vars/env_vars.html#ros-discovery-server) section in the Fast DDS documentation for further information.


Moreover, exporting that environment variable in the discovery server shell would make it try to connect it to other discovery server with that configuration.
In order to configure a discovery server to use TCP at a specific IP address and port, this is the proper command:
```bash
fastdds discovery -i 0 -t 127.0.0.1 -q 12345
```
Run `fastdds discovery -h` to check the other possible configurations

**Comment by JesusPoderoso** [comment]
As this is not an issue, but a support question, I am moving this thread to the support discussion forum according to the Fast DDS [contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support).
